plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/MenuContributionInitializer.java

```diff
@@ -16,16 +16,20 @@
 
 package com.google.cloud.tools.eclipse.login.ui;
 
+import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
+import com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory;
 import org.eclipse.ui.IStartup;
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.commands.ICommandService;
 
 public class MenuContributionInitializer implements IStartup {
-
+  IGoogleApiFactory apiFactory = new GoogleApiFactory();
+  
   @Override
   public void earlyStartup() {
     updateLoginCommand();
+    apiFactory.addCredentialChangeListener(() -> updateLoginCommand());
```

Comment:
Is there an on-menu-open event that can be used to query credential status only when opened, or does the menu have to be pre-calculated and proactively updated like this?

Reply:
I'm not entirely sure if this is possible. We have this `plugin.xml` defining the extension  https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/cf10c1047efe35077565216274f909f938eb0466/plugins/com.google.cloud.tools.eclipse.login/plugin.xml#L23-L52

It tells it the `GoogleLoginCommandHandler` how to update the element and what it does when clicked in three points (File menu, Google Cloud menu and lower right status bar). These are three elements (of probably different types) - this is just a conjecture in favor of a hook of this kind not being available.

I'll keep looking at the docs 

[menu extensions](https://download.eclipse.org/rt/rap/doc/3.6/guide/reference/extension-points/org_eclipse_ui_menus.html) for reference

Reply:
Here is a [list of examples](https://wiki.eclipse.org/Menu_Contributions) of `menuContribution` - it seems like there is no way to subscribe to dropdown opening events. `updateElement` gets called even if the element is closed, and does not get called when the dropdown is opened/closed.

Reply:
Thanks for looking into this.